### PR TITLE
Add listner and example how to get information about saved consents

### PR DESCRIPTION
**Pull request by Grzelazny**
New RingPublishingGDPRListener can be used to initialize libraries on ech place inside application.

### CODE REVIEW

File: demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java [hunk 1]
```diff
@@ -57,6 +57,24 @@ Otherwise you must move initialization libraries to section like showAppContent(
         // When (in you usage context) GDPR does not apply, use initialize method with extra parameter gdprApplies set to false
         // ringPublishingGDPR.initialize(false, this, appTenantId, appBrandingName, ringPublishingGDPRUIConfig);
 
+        // Optional listener that inform application about saving or updating consents.
```
Comment (pkordal) [inline demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java:60]: informs

File: demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java [hunk 2]
```diff
@@ -57,6 +57,24 @@ Otherwise you must move initialization libraries to section like showAppContent(
...
+        // You can add more listeners on each place where you initialize sdk.
```
Comment (pkordal) [inline demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java:61]: SDK

File: demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java [hunk 3]
```diff
@@ -39,6 +41,17 @@ protected void onCreate(Bundle savedInstanceState)
             final Intent startSettingScreenIntent = RingPublishingGDPR.getInstance().createShowSettingsScreenIntent(this);
             startActivity(startSettingScreenIntent);
         });
+
+        // Optional example: You can add consentsUpdate listener where you want
+        // Remember remove listned on object ondestroy
```
Comment (pkordal) [inline demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java:46]: Remember to remove / Remember about removing
ondestroy callback?

File: demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java [hunk 4]
```diff
@@ -61,4 +74,22 @@ public void onBackPressed()
         finish(); // Just for this example to not open Splash Screen when back from MainActivity
         super.onBackPressed();
     }
+
+    @Override
+    public void onConsentsUpdated()
+    {
+        // Here you can check if user agreed for all available vendors
```
Comment (pkordal) [inline demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java:81]: agreed on all..

File: demo/src/main/java/com/ringpublishing/gdpr/demo/SplashActivity.java [hunk 5]
```diff
@@ -56,28 +57,31 @@ private void showAppContent()
...
-        // you can use this flag to check if you can enable / initialize them as user agreed for all vendors
+
     }
 
-    private void initializeExternalLibraries()
+    private void initializeSplashExternalLibraries()
     {
-        // Here you can initialize external libraries, because consent screen was already visited by user.
+        // Here you can initialize external libraries initialized in your Splash screen,
+        // because consent screen was already visited by user.
```
Comment (pkordal) [inline demo/src/main/java/com/ringpublishing/gdpr/demo/SplashActivity.java:72]: consent screen has been already presented to user.

File: RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/RingPublishingGDPR.java [hunk 6]
```diff
@@ -215,6 +220,24 @@ public Intent createShowSettingsScreenIntent(Context context)
         return RingPublishingGDPRActivity.createShowSettingsScreenIntent(context);
     }
 
+    public void addRingPublishingGDPRListeners(RingPublishingGDPRListener ringPublishingGDPRListener)
```
Comment (pkordal) [inline RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/RingPublishingGDPR.java:223]: tu i niżej nie masz java doca

File: demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java (hunk 1 above)
Comment (Grzelazny) [reply demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java:60]: Changed

File: demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java (hunk 2 above)
Comment (Grzelazny) [reply demo/src/main/java/com/ringpublishing/gdpr/demo/DemoApplication.java:61]: Changed

File: demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java (hunk 3 above)
Comment (Grzelazny) [reply demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java:46]: Changed

File: demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java (hunk 4 above)
Comment (Grzelazny) [reply demo/src/main/java/com/ringpublishing/gdpr/demo/MainActivity.java:81]: Changed

File: demo/src/main/java/com/ringpublishing/gdpr/demo/SplashActivity.java (hunk 5 above)
Comment (Grzelazny) [reply demo/src/main/java/com/ringpublishing/gdpr/demo/SplashActivity.java:72]: Changed

File: RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/RingPublishingGDPR.java (hunk 6 above)
Comment (Grzelazny) [reply RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/RingPublishingGDPR.java:223]: Added

**Review by map-android — APPROVED** [review]

**Pull request merged**